waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -267,8 +267,12 @@ method getAll*(db: WakuMessageStore, onData: message_store.DataProc): MessageSto
   ok gotMessages
 
 proc adjustDbPageSize(dbPageSize: uint64, matchCount: uint64, returnPageSize: uint64): uint64 {.inline.} =
-  var ret = if matchCount < 2: dbPageSize * returnPageSize
+  const maxDbPageSize: uint64 = 20000 # the maximum DB page size is limited to prevent excessive use of memory in case of very sparse or non-matching filters. TODO: dynamic, adjust to available memory
+  if dbPageSize >= maxDbPageSize: return maxDbPageSize
```

Comment:
I prefer that we put early returns in different lines. This one-liner does not help with code readability.
```suggestion
  if dbPageSize >= maxDbPageSize: 
    return maxDbPageSize
```